src/commonMain/kotlin/baaahs/sim/FakeNetwork.kt

```diff
@@ -111,6 +120,69 @@ class FakeNetwork(
             }
         }
 
+        private val mdns = FakeMdns()
+
+        inner class FakeMdns() : Network.Mdns {
+
+            override fun register(
+                hostname: String,
+                type: String,
+                proto: String,
+                port: Int,
+                domain: String,
+                params: MutableMap<String, String>
+            ): Network.MdnsRegisteredService? {
+                val fullname = "$hostname.$type.$proto.${domain.normalizeMdnsDomain()}"
+                val inst = FakeRegisteredService(hostname, type, proto, port, domain.normalizeMdnsDomain(), params)
+                services[fullname] = inst
+                (inst as? FakeRegisteredService)?.announceResolved()
+                return inst
+            }
+
+            override fun unregister(inst: Network.MdnsRegisteredService?) {
+                if (inst != null) {
+                    val fullname = "${inst.hostname}.${inst.type}.${inst.proto}.${inst.domain}"
+                    (services.remove(fullname) as? FakeRegisteredService)?.announceRemoved()
```

Comment:
Cast probably not needed.

Reply:
This is the one that, ideally, should fail when caller didn’t register it. But I have the inst (which only the registering component should have in the first place), maybe I move this functionality to the MdnsRegisteredService implementation, and this method just calls `inst.unregister()`?